### DISCUSSION TITLE
Easy Start Guide?

### DISCUSSION THREAD

**Discussion by GalinMcMahon** [JPersson77/LGTVCompanion#245] [General]
I think I've found what I'm looking for!  I'm just trying to change my LG inputs from my computer via a Stream Deck.  Unfortunately I'm only "Philly" tech savvy so most of the instructions are over my head.

I saw a YouTube video of some guy who never learned how to speak politely and he set this up seemingly very fast.  But going through the git pages, it looks like there are a ton of steps.  Is there maybe a tutorial video somewhere of a full install and setup geared towards people who are not programmers?  Or maybe an A-Z install guide that starts with "Press the big button on the computer that says 'Power.'" (OK, maybe not THAT basic but some hand holding is good.)

**Comment by JPersson77** [comment]
Hi @GalinMcMahon it is a good idea. I'm not very much into video production myself but let's keep the discussion up and see if someone is interested in producing an instructional video and I will post it in the instructions.

There is not really a ton of steps but there are a few. The most difficult is to assign a static IP (fixed) for the TV in your routers setup. This is needed  to avoid a potential future situation whete the IP of the TV is reallocated by the router and invalidating the configuration.

Once the TV is setup it is a matter of executing a command line instruction to change the HDMI input: 

`"c:/program files/lgtv companion/lgtv companion.exe" -setHdmi 2`

Can  you try following instructions for setup here and let me know when/if you get stuck: [link](https://github.com/JPersson77/LGTVCompanion#lgtv-companion)

**Comment by GalinMcMahon** [comment]
Thanks a million for taking the time to reply!  If someone sees this and decides to make that tutorial, please also explain how to execute a command line instruction.  LOL sorry it's probably super obvious to most here but not to this guy.  I know WHAT that is so high five.

I will follow your link tonight and report back.  I'd be happy to "dumb it down" for future versions of me if I do get through it.  Thanks again for your help and for making this software :)

**Comment by GalinMcMahon97005** [comment]
OK.  I installed the program and it worked with ease.  I have moved on to the command line documentation.  This is where the layman probably gets lost.  Where do we enter the commands?  I tried using cmd and then some of the sample lines but got '"LGTV Companion.exe"' is not recognized as an internal or external command, operable program or batch file.  The documentation says that the programs "accept" command line arguments but there's no place to put them in the applet.  I checked the program files for a text file where these might be stored but there doesn't seem to be any files besides the applications and an ico file.

I understand this may be a little too much explanation but if it's manageable, I think it would help maybe some other people besides myself.  Hopefully anyway :)

**Comment by JPersson77** [comment]
No worries! 

A command line instruction is executed in a console. A console is a command line, text based, interface

To open the console on your PC, open the start menu and type "cmd" (without quotes). Click on "CMD.exe" and the console will open. 

Assuming you installed LGTVC in the default directory, you now need to change the working directory by typing

`cd "c:/program files/lgtv companion/"`

Press Enter and the prompt should change to indicate your working directory has changed.

You can now issue command line instructions to LGTVC by ttping your commands into the prompt like so:

`lgtvcli.exe -sethdmi 1`

Lgtvcli.exe is the executable program and the rest of the instruction are the parameters the program will execute. 

Please note that you can also send the exact same command line instructions to lgtv companion.exe which is useful and recommended when used in a script or in a shortcut for your stream deck. The difference is that with lgtvcli.exe a console window will pop up (it is a console application), but with lgtv companion.exe there will be no console window (it is a windows gui application). Try both in a stream deck shortcut to see what I mean.

Let me know how it goes
